src/viam/robot/client.py

```diff
@@ -185,22 +187,37 @@ async def refresh(self):
         """
         response: ResourceNamesResponse = await self._client.ResourceNames(ResourceNamesRequest())
         resource_names: List[ResourceName] = list(response.resources)
-        if resource_names == self._resource_names:
-            return
-        manager = ResourceManager()
-        for rname in resource_names:
-            if rname.type not in [RESOURCE_TYPE_COMPONENT, RESOURCE_TYPE_SERVICE]:
-                continue
-            if rname.subtype == "remote":
-                continue
-            try:
-                manager.register(Registry.lookup_subtype(Subtype.from_resource_name(rname)).create_rpc_client(rname.name, self._channel))
-            except ResourceNotFoundError:
-                pass
         with self._lock:
+            if resource_names == self._resource_names:
+                return
+            for rname in resource_names:
+                if rname.type not in [RESOURCE_TYPE_COMPONENT, RESOURCE_TYPE_SERVICE]:
+                    continue
+                if rname.subtype == "remote":
```

Comment:
should `remote` also be a constant?

Reply:
We only use it here, so I don't think it needs to be a constant yet. Once we use it elsewhere, then yes I think it should be a constant